src/twist_marker.cpp

```diff
@@ -107,21 +121,33 @@ class TwistMarkerPublisher : public rclcpp::Node
   {
     std::string frame_id;
     double scale;
+    bool use_stamped;
     double z;
 
     this->declare_parameter("frame_id", "base_footprint");
     this->declare_parameter("scale", 1.0);
+    this->declare_parameter("use_stamped", false);
     this->declare_parameter("vertical_position", 2.0);
 
     this->get_parameter<std::string>("frame_id", frame_id);
     this->get_parameter<double>("scale", scale);
+    this->get_parameter<bool>("use_stamped", use_stamped);
     this->get_parameter<double>("vertical_position", z);
 
     marker_ = std::make_shared<TwistMarker>(frame_id, scale, z);
 
-    sub_ = this->create_subscription<geometry_msgs::msg::Twist>(
-      "twist", rclcpp::SystemDefaultsQoS(),
-      std::bind(&TwistMarkerPublisher::callback, this, std::placeholders::_1));
+    if (use_stamped)
+    {
+      sub_stamped_ = this->create_subscription<geometry_msgs::msg::TwistStamped>(
+        "twist", rclcpp::SystemDefaultsQoS(),
+        std::bind(&TwistMarkerPublisher::callback_stamped, this, std::placeholders::_1));
+    }
+    else
+    {
+      sub_ = this->create_subscription<geometry_msgs::msg::Twist>(
+        "twist", rclcpp::SystemDefaultsQoS(),
+        std::bind(&TwistMarkerPublisher::callback, this, std::placeholders::_1));
+    }
```

Comment:
Nav2 uses (just) `Twist` for `humble`, so I'd propose the following:

- `humble`/`iron`: `Twist` subscription + both options for publishing with the parameter. No need of updating `joystick_relay` here.
- `jazzy`/`rolling`: All `TwistStamped`.

This would include branching off, as I commented before.